### PR TITLE
Processor: capture error/warning locations as well as the messages.

### PR DESCRIPTION
**Pull request by kriegfrj**
Make sure that when error() and warning() call location(), they do so on the current owner process instead of on themselves.

Includes a test case.

### CODE REVIEW

File: biz.aQute.bndlib.tests/src/test/ProcessorTest.java [hunk 1]
```diff
@@ -188,6 +191,44 @@ private void assertNative(String in, String filter, String... fixup) throws Exce
 		}
 	}
 
+	static class SeethroughProcessor extends Processor {
```
Comment (bjhargrave) [inline biz.aQute.bndlib.tests/src/test/ProcessorTest.java]: Not sure what the purpose of this subclass is. It does not change the behavior of Processor at all. I think you can remove this subclass.

File: biz.aQute.bndlib/src/aQute/bnd/osgi/Processor.java [hunk 2]
```diff
@@ -290,7 +290,7 @@ public SetLocation warning(String string, Object... args) {
 		if (!p.warnings.contains(s))
 			p.warnings.add(s);
 		p.signal();
-		return location(s);
+		return p.location(s);
```
Comment (bjhargrave) [inline biz.aQute.bndlib/src/aQute/bnd/osgi/Processor.java]: Great job finding this subtle bug!

File: biz.aQute.bndlib.tests/src/test/ProcessorTest.java (hunk 1 above)
Comment (kriegfrj) [reply biz.aQute.bndlib.tests/src/test/ProcessorTest.java]: The accessibility on `Processor.beginHandleErrors()`/`endHandleErrors()` is protected. We can't invoke these methods from within the test method on a bare `Processor` object. Subclassing it was the easiest way to gain direct access to the methods for testing, as subclasses are allowed to widen the accessibility of overridden methods (hence calling it "Seethrough" - you can see the methods that ordinarily would be).

To be fair, this is not immediately obvious. If it helps, I can add a comment to this effect to explain to the next developer why this was done.

Another alternative would be to use reflection to bypass the accessibility check, but unless you have a good reflection helper API on the test classpath this code wouldn't be much easier to read.

Comment (bjhargrave) [reply biz.aQute.bndlib.tests/src/test/ProcessorTest.java]: Of course. I did not see that. I guess a comment would help to explain. Thanks for educating me!

Comment (kriegfrj) [reply biz.aQute.bndlib.tests/src/test/ProcessorTest.java]: Actually, perhaps a better alternative would be to move the unit test into the `aQute.bnd.osgi` package, and then the protected methods would be accessible from the test method without me needing to override it. I'm happy to implement whatever you suggest.

File: biz.aQute.bndlib/src/aQute/bnd/osgi/Processor.java (hunk 2 above)
Comment (kriegfrj) [reply biz.aQute.bndlib/src/aQute/bnd/osgi/Processor.java]: It's small fries compared with what you guys have done on the rest of the project, so I don't really have a right to accolades.

File: biz.aQute.bndlib.tests/src/test/ProcessorTest.java (hunk 1 above)
Comment (bjhargrave) [reply biz.aQute.bndlib.tests/src/test/ProcessorTest.java]: Use the subclass with a comment to explain why.

**Comment by kriegfrj** [comment]
I've added the comment to the test class. Let me know if you have any other suggestions.

**Comment by bjhargrave** [comment]
> I've added the comment to the test class.

Are you sure you pushed it? I don't see the comment in the commit in the PR.

**Comment by kriegfrj** [comment]
Well, it looks like I pushed _something_ - but apparently it didn't include any changes! I must have rebased and then forgotten to actually commit the change or something. Apologies.

Should be there now.

**Pull request merged**